python/pyspark/ml/tuning.py

```diff
@@ -143,6 +145,8 @@ def getEvaluator(self):
 
 class CrossValidator(Estimator, ValidatorParams):
     """
+    .. note:: Experimental
```

Comment:
Right now the Scala doc in ml is :-1: 
>  
> * K-means clustering with support for k-means|| initialization proposed by Bahmani et al.
>  *
>  * @see [[http://dx.doi.org/10.14778/2180912.2180915 Bahmani et al., Scalable k-means++.]]
> 

The mllib one is:
 

> * K-means clustering with a k-means++ like initialization mode
>  * (the k-means|| algorithm by Bahmani et al).
>  *
>  * This is an iterative algorithm that will make multiple passes over the data, so any RDDs given
>  * to it should be cached by the user.
> 

I'm happy to try write a longer  scaladoc on KMeans or we could just copy the link over to match Scala - whats your preference?